docs/user/openstack/install_upi.md

```diff
@@ -10,7 +10,57 @@ The installer is still used to generate the ignition files and monitor the insta
+
+- [Installing OpenShift on OpenStack User-Provisioned Infrastructure](#installing-openshift-on-openstack-user-provisioned-infrastructure)
+  - [Table of Contents](#table-of-contents)
+  - [Prerequisites](#prerequisites)
+  - [OpenShift Configuration Directory](#openshift-configuration-directory)
+    - [Red Hat Enterprise Linux CoreOS (RHCOS)](#red-hat-enterprise-linux-coreos-rhcos)
+  - [API and Ingress Floating IP Addresses](#api-and-ingress-floating-ip-addresses)
+  - [Install Config](#install-config)
+    - [Fix the Node Subnet](#fix-the-node-subnet)
+    - [Empty Compute Pools](#empty-compute-pools)
+    - [Modify NetworkType (Required for Kuryr SDN)](#modify-networktype-required-for-kuryr-sdn)
+  - [Edit Manifests](#edit-manifests)
+    - [Remove Machines and MachineSets](#remove-machines-and-machineSets)
+    - [Make Control-Plane Nodes Unschedulable](#make-control-plane-nodes-unschedulable)
+  - [Ignition Config](#ignition-config)
+    - [Infra ID](#infra-id)
+    - [Update Bootstrap Ignition](#update-bootstrap-ignition)
+    - [Update Master Ignition](#update-master-ignition)
+  - [Network Topology](#network-topology)
+    - [Network and Subnet](#network-and-subnet)
+    - [Subnet DNS (optional)](#subnet-dns-optional)
+    - [Security Groups](#security-groups)
+      - [Master and Worker Security Groups](#master-and-worker-security-groups)
+      - [Master Security Group Rules](#master-security-group-rules)
+      - [Worker Security Group Rules](#worker-security-group-rules)
+    - [Router](#router)
+    - [Public API and Ingress Access](#public-api-and-ingress-access)
+      - [API and Ingress Ports](#api-and-ingress-ports)
+      - [Assign Floating IPs to the Ports](#asssign-floating-ips-to-the-ports)
+      - [Create Public DNS Records](#create-public-dns-records)
+  - [Bootstrap](#bootstrap)
+    - [Bootstrap Ignition](#bootstrap-ignition)
+    - [Bootstrap Ignition Shim](#bootstrap-ignition-shim)
+    - [Bootstrap Port](#bootstrap-port)
+    - [Bootstrap Server](#bootstrap-server)
+  - [Control Plane](#control-plane)
+    - [Control Plane Ports](#control-plane-ports)
+    - [Control Plane Trunks (Required for Kuryr SDN)](#control-plane-trunks-required-for-kuryr-sdn)
+    - [Control Plane Servers](#control-plane-servers)
+    - [Wait for the Control Plane to Complete](#wait-for-the-control-plane-to-complete)
+    - [Access the OpenShift API](#access-the-openshift-api)
+    - [Delete the Bootstrap Resources](#delete-the-bootstrap-resources)
+  - [Compute Nodes](#compute-nodes)
+    - [Compute Nodes ports](#compute-nodes-ports)
+    - [Compute Nodes Trunks (Required for Kuryr SDN)](#compute-nodes-trunks-required-for-kuryr-sdn)
+    - [Compute Nodes Servers](#compute-nodes-servers)
+    - [Approve the worker CSRs](#approve-the-worker-csrs)
+    - [Wait for the OpenShift Installation to Complete](#wait-for-the-openshift-installation-to-complete)
+  - [Destroy the OpenShift Cluster](#destroy-the-openshift-cluster)
+  
 ## Prerequisites
```

Comment:
"This document uses a cloud named openstack"
I think this is confusing

Reply:
I think the way we are using this section to explain the examples we use in the doc, rather than the actual pre-requisite requirements that the user might need is very confusing.

Reply:
--> perhaps this should be a separate commit/pull

Reply:
I added "The values that we use as examples in the document may vary in different OpenStack deployments. So, please consult with your cloud administrator to clarify them."

Reply:
I think that is fine for now, the changes you made are good, so I wont bog you down with this. I think we should create a separate card for revising and clarifying the docs though.